File: backend/src/Models/Message.cs
namespace backend.Models

{
    public class Message
    {
        public int Id { get; set; }
        public string Content { get; set; } = String.Empty;
        public int ReceiverId { get; set; }
        public User? Receiver { get; set; }
        public int SenderId { get; set; }
        public User? Sender { get; set; }
        public int TeamId { get; set; }
        public Team? Team { get; set; }
    }
}

File: frontend/src/utils/requests/Message.ts
import axios from "axios";

export interface Message {
    id: number;
    content: string;
    receiverId: number;
    senderId: number;
    teamId: number;
}

export const createMessage = async (content: string, receiverName: string, senderName: string, teamName: string) => {
    return axios
        .post(`http://localhost:5002/api/Message`, {
            content,
            receiverName,
            senderName,
            teamName
        }, {
            headers: {
                Authorization: `Bearer ${localStorage.getItem("token")}`,
            }
        })
        .then((response) => {
            return response.data;
        });
}

export const getUserMessages = async (userId: number) => {
    return axios
        .get(`http://localhost:5002/api/Message/${userId}`, {
            headers: {
                Authorization: `Bearer ${localStorage.getItem("token")}`,
            }
        })
        .then((response) => {
            return response.data;
        });
}

export const deleteMessage = async (id: number) => {
    return axios
        .delete(`http://localhost:5002/api/Message/${id}`, {
            headers: {
                Authorization: `Bearer ${localStorage.getItem("token")}`,
            }
        })
        .then((response) => {
            return response.data;
        });
}